File: AudioTransfer/AudioTransfer.Core/Network/UdpClientSession.cs
using System;
using System.Net;

namespace AudioTransfer.Core.Network
{
    public enum HandshakeState
    {
        None,
        SynReceived,
        Authenticated
    }

    public class UdpClientSession
    {
        public IPEndPoint EndPoint { get; }
        public HandshakeState State { get; set; } = HandshakeState.None;
        public DateTime LastSeenUtc { get; set; } = DateTime.UtcNow;
        public string? DeviceName { get; set; }

        public UdpClientSession(IPEndPoint endPoint)
        {
            EndPoint = endPoint;
        }

        public bool IsActive(double timeoutSeconds = 10)
        {
            return (DateTime.UtcNow - LastSeenUtc).TotalSeconds < timeoutSeconds;
        }
    }
}

File: AudioOverLAN/app/src/main/java/com/example/audiooverlan/network/UdpClientSession.java
package com.example.audiooverlan.network;

import java.net.InetAddress;

public class UdpClientSession {
    public enum HandshakeState {
        Disconnected,
        SynReceived,
        Authenticated
    }

    private final InetAddress address;
    private final int port;
    private HandshakeState state;
    private long lastSeenMillis;

    public UdpClientSession(InetAddress address, int port) {
        this.address = address;
        this.port = port;
        this.state = HandshakeState.Disconnected;
        this.lastSeenMillis = System.currentTimeMillis();
    }

    public InetAddress getAddress() {
        return address;
    }

    public int getPort() {
        return port;
    }

    public HandshakeState getState() {
        return state;
    }

    public void setState(HandshakeState state) {
        this.state = state;
    }

    public void updateLastSeen() {
        this.lastSeenMillis = System.currentTimeMillis();
    }

    public boolean isActive() {
        return (System.currentTimeMillis() - lastSeenMillis) < 5000;
    }
}
